.changeset/dull-carrots-matter.md

```diff
@@ -0,0 +1,10 @@
+---
+"@pnpm/default-resolver": minor
+"@pnpm/tarball-resolver": minor
+"@pnpm/local-resolver": major
+"@pnpm/resolver-base": major
```

Comment:
`major` because `WorkspaceResolveResult` has been moved from `resolver-base` to `npm-resolver`.